File: level1/07-anagrams/main.rb



# def sorted_string(s)
#   arr = []
  
#   s.downcase.each_byte.to_a.map { |e| 
#     if e >= 0 and e <= 9  or e >= 97 and e <= 122
#       arr << e
#     end
#   }
  
#   mergeSort(arr).map{ |e| e.chr }.join('')
# end

def sort_string_with_point(s)
  arr = []
  s.each_byte do |e|
    # downcase
    if e >= 65 and e <= 90
      e += 32
    end

    # accept alphanumeric
    if e >= 0 and e <= 9  or e >= 97 and e <= 122
      arr << e
    end
  end
  arr.sum
end

def anagrams(s1, s2)
  sort_string_with_point(s1) == sort_string_with_point(s2)
end

p anagrams('rail safety', 'fairy tales')
p anagrams('RAIL! SAFETY!', 'fairy tales')
p anagrams('Hi there', 'Bye there')

File: level1/07-anagrams/main.go
package main

import (
	"fmt"
)

func anagrams(s1 string, s2 string) bool {
	return score(s1) == score(s2)
}

func score(s string) int {
	// var arr []byte
	sBytes := []byte(s)
	score := 0
	for _, e := range sBytes {
		// downcase
		if e >= 65 && e <= 90 {
			e = e + 32
		}
		//accept alphanumeric
		if (e >= 0 && e <= 9) || (e >= 97 && e <= 122) {
			score = score + int(e)
		}
	}
	return score
}

func main() {
	fmt.Println(anagrams("rail safety", "fairy tales"))
	fmt.Println(anagrams("RAIL! SAFETY!", "fairy tales"))
}
